### PR TITLE
Bring missing changes from myrra repo

### PR DESCRIPTION
**Pull request by mFragaBA**
On #44 we brought the game server from the curse of myrra repo to this one. 

I ran `git log --pretty=oneline -- server` on the curse_of_myrra repo to check if there were some PRs that got merged to the curse of myrra repo and were not merged on time to #44:

```
df50c3e04e8d2f85e9b5331524096d3edf9ab461 (HEAD -> main, origin/main, origin/HEAD) improve players balance (#1188)
a3c3b1661dba1587316f89d55822e1ab62461010 Auto attacks (#1163)
660e0ff7fcf5500f3241021c78c23709905560e4 More metrics (#1162)
cd105df2ba353e8eb50f628275b5b1fe02f68341 Gh 1125 load tests (#1132)
97d12a1097c5b9217d1150aa422675c76098993e [GH-1126] Fix end game backend bugs (#1137)
8ab524871145181bce5a5afa06d60ea3d9c88330 Zone Killfeed (#1159)
da6d0c892c720898b59103153e99b740dd371c67 (custom_api_for_load_tests) New flow + New Engine (#1087)
```

All of those but #1162 and #1132 got merged. This PR brings those changes:

- [bring changes from #1132](https://github.com/lambdaclass/game_backend/pull/50/commits/96231aeb03e7f5348730efea06f3916e1f37fb5d)
- [bring changes from #1162](https://github.com/lambdaclass/game_backend/pull/50/commits/96e1faa08f210ccdabadaf3efd40ee064634b80e)
- On the remaining commits I fixed an issue introduced in #44 on the generation of protobuf files for the load test
- Removed not used code in [5731ccd](https://github.com/lambdaclass/game_backend/pull/50/commits/5731ccd6de9464f056dea9733eee6c76dead0c16)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -57,10 +57,10 @@ gen-server-protobuf:
 
 gen-load-test-protobuf:
 	protoc \
-		--elixir_out=./load_test/lib/load_test/ \
+		--elixir_out=./load_test/lib/load_test/communication \
```
Comment (manucamejo) [inline Makefile:60]: Is this change necessary? Or is it working properly now?

File: load_test/README.md [hunk 2]
```diff
@@ -1,31 +1,45 @@
...
-If you want to see a request tracking report for every player of every game after a load test, you can run the helper function `DarkWorldsServer.RunnerSupervisor.RequestTracker.report/1` *on the server*. Running
+When the game server is ran as a systemd service you might need to edit the service config. You can check [LimitNOFILE on the server setup script](./setup_game_server.sh) although it's a one time thing. You shouldn't do it every time.
+
+## Analyzing results
+
+If you want to see a request tracking report for every player of every game after a load test, you can run the helper function `DarkWorldsServer.Engine.RequestTracker.report/1` *on the server*. Running
 
 ```
-DarkWorldsServer.RunnerSupervisor.RequestTracker.report(:game)
+DarkWorldsServer.Engine.RequestTracker.report(:game)
```
Comment (manucamejo) [inline load_test/README.md:42]: We don't want this change.

File: load_test/README.md [hunk 3]
```diff
@@ -49,7 +63,7 @@ Details per game
 while running
 
 ```
-DarkWorldsServer.RunnerSupervisor.RequestTracker.report(:player)
+DarkWorldsServer.Engine.RequestTracker.report(:player)
```
Comment (manucamejo) [inline load_test/README.md:66]: We don't want this change.

**Review by manucamejo — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (mFragaBA) [reply Makefile:60]: It is

**Review by fkrause98 — DISMISSED** [review]
LGTM! 🚀

**Review by fkrause98 — APPROVED** [review]
LGTM! 🚀

**Review by manucamejo — APPROVED** [review]
